### ISSUE TITLE
Setting status-position top on ssh nested causing issue with cursor position

### ISSUE THREAD

**Issue by budimanjojo** [tmux/tmux#1952]
### Issue description

If I start a nested tmux in ssh session with `set -g status-position top` the cursor will go on top of the prompt

### Required information

Please provide the following information:

* tmux version (`tmux -V`).
`tmux next-3.1`
* Platform (`uname -sp`).
`Linux x86_64`
`
* $TERM inside and outside of tmux (`echo $TERM`).
Inside tmux `tmux-256color`
Outside tmux `xterm-256color`
* Logs from tmux (`tmux kill-server; tmux -vv new`).
[tmux-server-20254.log](https://github.com/tmux/tmux/files/3747771/tmux-server-20254.log)

EDIT: Forgot to mention that my tmux-server is having `set -g status-position bottom`. Also forgot to attach a screenshot of it
![Screenshot from 2019-10-20 17-49-04](https://user-images.githubusercontent.com/13085918/67158693-96471c80-f365-11e9-947b-bcebe839e6ea.png)

**Comment by nicm** [comment]
I can't reproduce this, can you make a minimal config for each tmux server that shows the problem?

**Comment by budimanjojo** [comment]
@nicm It happens on me even if I have a minimal config on both server and client. How to reproduce:
1. Have an autostart tmux script on the ssh client. And create a `~/.tmux.conf` file with just this line: `set -g status-position top`.
2. On the server, start tmux and ssh into the client.

This is the logfile. 


[tmux-server-30078.log](https://github.com/tmux/tmux/files/3778079/tmux-server-30078.log)
![Screenshot from 2019-10-28 16-27-07](https://user-images.githubusercontent.com/13085918/67668664-f744a500-f9a2-11e9-904a-065e31ccfdb0.png)

**Comment by budimanjojo** [comment]
I can reproduce this everytime.. I just tried this on my vps server and it still happens

**Comment by nicm** [comment]
Are both tmux the same version?

**Comment by budimanjojo** [comment]
@nicm Okay that actually make sense. I was using tmux 2.8 on both servers.
But I just tried again on my home computer with tmux 2.9a and installed the same 2.9a on my server. And the problem is still there. Also, forgot to mention that I'm using zsh instead of bash.

**Comment by nicm** [comment]
Is this log from the outer or inner tmux?

If you do C-b r before pressing any other keys so the outer tmux redraws, does the prompt move to the right place?

If not, does C-b C-b r so the inner tmux redraws fix it?

**Comment by nicm** [comment]
Never mind the first bit, I can see the log is from the outer tmux now, that's OK.

**Comment by nicm** [comment]
OK can you please reproduce this in a terminal that is 80x24 in size and show me the log from the outer tmux? Use master for both tmux please.

**Comment by nicm** [comment]
If I copy what the inner tmux is sending it seems OK to me, but my terminal is smaller so I need the output from a smaller size terminal to test with.

**Comment by budimanjojo** [comment]
> Is this log from the outer or inner tmux?
> 
> If you do C-b r before pressing any other keys so the outer tmux redraws, does the prompt move to the right place?
> 
> If not, does C-b C-b r so the inner tmux redraws fix it?

Redrawing the inner tmux seems to fix it partially. It reposition the cursor on the prompt like it should be. But there is another prompt hanging below it like this:
![Screenshot from 2019-11-08 19-59-15](https://user-images.githubusercontent.com/13085918/68478453-7cad3c80-0262-11ea-8521-9298d7a5447b.png)

Sho here's the new log file with window size of 80x24. Blank `.tmux.conf` on the outer `tmux next-3.1` and `.tmux.conf` with only one line: `set -g status-position top` on the inner `tmux next-3.1`.
[tmux-server-26355.log](https://github.com/tmux/tmux/files/3824688/tmux-server-26355.log)

**Comment by budimanjojo** [comment]
Also, is it because I have a zsh prompt with two line?

Thanks before :smile:

**Comment by nicm** [comment]
Are you using GNOME terminal or another VTE-based terminal? It looks like there is a bug in VTE.

Does this still happen in xterm?

**Comment by nicm** [comment]
It looks like VTE is incorrectly handling cursor movement outside the scroll region, compare this in GNOME terminal and in xterm:
~~~~
printf '\033[r\033[H\033[2J'
printf '\033[4;7r\033[3;1Haaa\033[Axxx'
sleep 10

**Comment by budimanjojo** [comment]
@nicm Wow it actually is not happening in xterm. although those `printf` commands you mentioned produce similar result in both xterm and gnome-terminal.
I'll close this issue as this is not tmux problem. Do you have any link to the VTE issue for this particular bug? Or point me to where I can report this bug. Thanks again :+1:

**Comment by nicm** [comment]
Perhaps there is a different bug in your VTE version. What VTE and GNOME
terminal versions are you using?

tmux will send slightly different escape sequences in xterm because it
supports margins so it might be worth also making sure the problem doesn't
exist in rxvt as well.

There is a VTE issue tracker here:

https://gitlab.gnome.org/GNOME/vte/issues/

**Comment by nicm** [comment]
When you say similar results from the commands I have do you mean
/identical/ results? For me, in one the xxx is above the aaa and in the
other it is below.

**Comment by budimanjojo** [comment]
I tried them inside Tmux and it displays similar result. But if I try them outside Tmux then they display different result like you mentioned above.
I'm using Gnome-terminal on my work machine (right now) and Termite on my home machine. They both suffer this same problem. I just tried Alacritty and Kitty and the prompt is displaying properly.
So maybe I'll try to report to upstream VTE. Thanks a lot for your help. :+1:

**Comment by nicm** [comment]
OK sounds good, thanks.

**Comment by egmontkob** [comment]
Thanks for the investigation guys!

It's indeed a VTE bug: https://gitlab.gnome.org/GNOME/vte/issues/197.

**Comment by budimanjojo** [comment]
@egmontkob The patch you provided indeed fix the issue. Thanks! :+1:

**Comment by egmontkob** [comment]
Fix released in vte-0.58.3.
